imagebuildah/build.go

```diff
@@ -1135,8 +1136,20 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) error
 			defer stageExecutor.Delete()
 		}
 		if err := stageExecutor.Execute(ctx, stage.Builder, stage.Node); err != nil {
-			return err
+			lastErr = err
 		}
+
+		// Delete the successful intermediate containers if an error in the build
+		// process occurs and b.removeIntermediateCtrs is true.
+		if lastErr != nil {
+			if b.removeIntermediateCtrs {
+				if err := stageExecutor.deleteSuccessfulIntermediateCtrs(); err != nil {
+					logrus.Error(err)
```

Comment:
I would ignore errors here and log in delete

Reply:
Fixed